pkg/rancher-desktop/backend/backendHelper.ts

```diff
@@ -122,15 +125,31 @@ export default class BackendHelper {
     return engineName === ContainerEngine.MOBY && semver.gte(kubeVersion, '1.24.1') && semver.lte(kubeVersion, '1.24.3');
   }
 
+  static checkForLockedVersion(newVersion: semver.SemVer, cfg: BackendSettings, sv: SettingsValidator): void {
+    const [, errors] = sv.validateSettings(cfg as Settings, { kubernetes: { version: newVersion.raw } }, settingsImpl.getLockedSettings());
+
+    if (errors.length > 0) {
+      if (errors.some(err => err.match(/field '.*' is locked/))) {
+        throw new LockedFieldError(`Error in deployment profiles:\n${ errors.join('\n') }`);
+      } else {
+        throw new Error(`Validation errors for requested version ${ newVersion }: ${ errors.join('\n') }`);
+      }
+    }
+  }
+
   /**
    * Validate the cfg.kubernetes.version string
    * If it's valid and available, use it.
    * Otherwise fall back to the first (recommended) available version.
    */
-  static async getDesiredVersion(currentConfigVersionString: string|undefined, availableVersions: semver.SemVer[], noModalDialogs: boolean, settingsWriter: (_: any) => void): Promise<semver.SemVer> {
+  static async getDesiredVersion(cfg: BackendSettings, availableVersions: semver.SemVer[], noModalDialogs: boolean, settingsWriter: (_: any) => void): Promise<semver.SemVer> {
+    const currentConfigVersionString = cfg?.kubernetes?.version;
     let storedVersion: semver.SemVer|null;
     let matchedVersion: semver.SemVer|undefined;
     const invalidK8sVersionMainMessage = `Requested kubernetes version '${ currentConfigVersionString }' is not a valid version.`;
+    const sv = new SettingsValidator();
+    const lockedSettings = settingsImpl.getLockedSettings();
+    const versionIsLocked = lockedSettings.kubernetes?.version ?? false;
```

Comment:
Would we want to convert `lockedSettings.kubernetes.version` to its boolean representation via something like `    const versionIsLocked = !!lockedSettings.kubernetes?.version ?? false;`? I know that truthy and falsy values are entirely valid, but doing the conversion more explicitly communicates what we are planning to do with this value later on.

Reply:
That isn't necessary. While the type of `lockedSettings` is  `Record<string, any>`, it's like that because I still never get recursive typescript types right -- it should be

```
type LockedSettingsType = Record<string, boolean | LockedSettingsType>
```

So all references are either boolean or null (the expression isn't represented in the current object), and the expression as above is always boolean without needing a `!!`.